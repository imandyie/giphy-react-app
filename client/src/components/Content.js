import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
import '../styles/components/Content.scss';
import colors from '../styles/colors';

export default class Content extends PureComponent {
  render() {
    const {
      text,
      color,
      opacity,
      className,
      isFetching,
      type,
    } = this.props;
    const customClassName = className || '';
    const fetchingClass = isFetching ? 'animated-content-placeholder' : '';

    return (
      <div
        className={`content ${customClassName} ${fetchingClass}`}
        style={{color, opacity}}
      >
        {type === 'link' ?
        (
          <a
            href={text}
            target="_blank"
          >
            {text}
          </a>
        ) : text}
      </div>
    );
  }
}

Content.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string,
  opacity: PropTypes.string,
  className: PropTypes.string,
  isFetching: PropTypes.bool,
  type: PropTypes.string,
};
