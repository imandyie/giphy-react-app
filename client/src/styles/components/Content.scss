@import '../colors.scss';

.content {
  margin: 0;
  padding: 0;
  font-weight: lighter;
  font-size: 0.8rem;
}

@keyframes contentPlaceholder {
  0% {
    background-position: -1200px 0
  }
  100% {
    background-position: 1200px 0
  }
}

.animated-content-placeholder {
  animation-duration: 2.1s;
  animation-fill-mode: forwards;
  animation-iteration-count: infinite;
  animation-name: contentPlaceholder;
  animation-timing-function: linear;
  background: $white;
  background: linear-gradient(to right, $darkerGray 8%, $darkGray 18%, $darkerGray 33%);
  background-size: 800px 104px;
  height: 80px;
  position: relative;
  max-width: 1200px;
  font-size: 0px;
}