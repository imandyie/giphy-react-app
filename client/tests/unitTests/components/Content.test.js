import React from 'react';
import Content from '../../../src/components/Content';
import renderer from 'react-test-renderer';

test('Content renders without errors', () => {
  const component = renderer.create(
    <Content
      text="Sample Content"
      color="red"
      opacity="1"
      className="sample-class"
      isFetching={false}
    />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Content renders as a link and class name is set during fetching', () => {
  const component = renderer.create(
    <Content
      text="Sample Content"
      color="red"
      opacity="1"
      className="sample-class"
      isFetching={true}
      type="link"
    />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});