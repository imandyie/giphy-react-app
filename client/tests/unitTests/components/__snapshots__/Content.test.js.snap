// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders as a link and class name is set during fetching 1`] = `
<div
  className="content sample-class animated-content-placeholder"
  style={
    Object {
      "color": "red",
      "opacity": "1",
    }
  }
>
  <a
    href="Sample Content"
    target="_blank"
  >
    Sample Content
  </a>
</div>
`;

exports[`Content renders without errors 1`] = `
<div
  className="content sample-class "
  style={
    Object {
      "color": "red",
      "opacity": "1",
    }
  }
>
  Sample Content
</div>
`;
